components/settings/tabs/Account.tsx

```diff
@@ -0,0 +1,442 @@
+import {
+  updateUserProfile,
+  updateUserAvatar,
+  deleteUserAvatar,
+} from '@/actions/user';
+import { profileConfigurationScheme } from '@/schemas';
+import {
+  Alert,
+  Anchor,
+  Avatar,
+  Button,
+  Fieldset,
+  Group,
+  InputBase,
+  Loader,
+  LoadingOverlay,
+  Stack,
+  TagsInput,
+  Text,
+  TextInput,
+  Textarea,
+  Tooltip,
+} from '@mantine/core';
+import { useForm, Form, yupResolver } from '@mantine/form';
+import { useDisclosure } from '@mantine/hooks';
+import { notifications } from '@mantine/notifications';
+import { IMaskInput } from 'react-imask';
+import {
+  IconBrandGithub,
+  IconBrandLinkedin,
+  IconCheck,
+  IconCheckbox,
+  IconFileUpload,
+  IconNotesOff,
+  IconX,
+} from '@tabler/icons-react';
+import { useSession } from 'next-auth/react';
+import React, { useRef, useState } from 'react';
+
+interface FormValues {
+  firstName?: string;
+  lastName?: string;
+  phone?: string;
+  school?: string;
+  bio?: string;
+  githubURL?: string;
+  linkedinURL?: string;
+  skills: string[];
+}
+
+export default function Account() {
+  // Component level states
+  const { data: session, status, update } = useSession();
+  const [isLoading, { open, close }] = useDisclosure();
+
+  // Avatar states, loading, and values
+  const inputFileRef = useRef<HTMLInputElement>(null);
+  const [loadingAvatar, setLoadingAvatar] = useState<boolean>(false);
+  const [isAvatarHovered, setIsAvatarHovered] = useState<boolean>(false);
+  const MAX_IMAGE_SIZE_MB = 4.5;
+  const MAX_FILE_SIZE_BYTES = 1000000 * MAX_IMAGE_SIZE_MB;
+
+  // Form states for editables
+  const [bioLength, setBioLength] = useState<number>(
+    session?.user?.bio?.length || 0
+  );
+  const [currentSkills, setCurrentSkills] = useState<string[]>(
+    session?.user?.skills || []
+  );
+  const [currentSkillError, setcurrentSkillError] = useState<boolean>(false);
+  const [formChanged, setFormChanged] = useState<boolean | null>(null);
+
+  const displayPhone = (phone: string) => {
+    if (!phone || phone.length !== 10) return null;
+
+    return `(${phone.slice(0, 3)})-${phone.slice(3, 6)}-${phone.slice(6)}`;
+  };
+
+  // Form initilization
+  const form = useForm({
+    mode: 'uncontrolled',
+    initialValues: {
+      bio: session?.user?.bio || '',
+      skills: currentSkills,
+    },
+    validate: yupResolver(profileConfigurationScheme),
+    onValuesChange: (values: any) => {
+      console.log(values);
+      if (values.bio !== null) setBioLength(values.bio.length);
+      setFormChanged(true);
+    },
+  });
+
+  // Preprocessing form values after submit
+  const cleanFormValues = (values: FormValues): Partial<FormValues> => {
```

Comment:
You can add `transformValues: (vals) => ...` in the `useForm` hook instead. TS is not happy even though ik what you're doing.